contract_sale_invoicing/models/contract.py

```diff
@@ -28,15 +28,10 @@ def _recurring_create_invoice(self, date_ref=False):
                         contract.partner_id.commercial_partner_id.ids,
                     ),
                     ("invoice_status", "=", "to invoice"),
-                    (
-                        "date_order",
-                        "<=",
-                        "{} 23:59:59".format(contract.recurring_next_date),
-                    ),
+                    ("date_order", ">=", contract.recurring_next_date),
                 ]
             )
             if sales:
-                invoice_ids = sales.action_invoice_create()
-                invoices |= self.env["account.invoice"].browse(invoice_ids)[:1]
-
-        return invoices
+                invoice_ids = sales._create_invoices()
+                invoices |= invoice_ids
+        return invoices[:1]
```

Comment:
Why returning only one record?

Reply:
@ThomasBinsfeld  if this way is incorrect then could you please give a better suggestion.

Reply:
> Why returning only one record?
When this method is called in the method "recurring_create_invoice" it shows singletone error if multiple invoices are passed.

Reply:
Simply don't change the previous code.

It returned a recordset with all generated invoices.